Удаление покупателя

MaksimovMA opened the pull request:
Проверить, что при удалении покупателя удаляются и записи всех связанных сущностей (списки избранного, товары в корзине, список заказов). А вот комменты не удалять, но продумать как сделать - выводить дефолтную картинку-аватар и в качестве имени, например, deleted.

В DataInitializer  добавлена новая сущность "DeletedCustomer" с ролью "ROLE_CUSTOMER", она встает на место удаленного пользователя в комментариях к товару.

MaksimovMA committed:
Добавляю кнопку удаления покупателя;
Добавляю нового пользователя который заменяет удаленных пользователях в комментариях тем самым сохраняя комментарии.

MaksimovMA committed:
Добавляю имя удаленному пользователю (Deleted)

MaksimovMA committed:
Добавляю имя удаленному пользователю (Deleted)

MaksimovMA committed:
Merge branch 'dev' into refactor/delete_customer

ghost commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Удаление покупателя](https://trello.com/c/tBVAiMvw/195-%D1%83%D0%B4%D0%B0%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BF%D0%BE%D0%BA%D1%83%D0%BF%D0%B0%D1%82%D0%B5%D0%BB%D1%8F)

MaksimovMA committed:
в application.yml прописал дефолтные настройки для username и password

MaksimovMA committed:
в application.yml прописал дефолтные настройки для username и password

MaksimovMA committed:
Merge branch 'dev' into refactor/delete_customer

MaksimovMA committed:
в application.yml прописал дефолтные настройки для username и password

MaksimovMA committed:
Изменяю маппинг ("/deleteProfile2/{id}") на ("/deleteProfileUnrecoverable/{id}")
Добавляю описание к методу deleteProfileUnrecoverable  в CustomerRestController

MaksimovMA committed:
Изменяю маппинг ("/deleteProfile2/{id}") на ("/deleteProfileUnrecoverable/{id}")
Добавляю описание к методу deleteProfileUnrecoverable  в CustomerRestController

tediik commented on src/main/java/com/jm/online_store/config/DataInitializer.java:
```diff
@@ -168,13 +168,19 @@ private void roleInit() {
         Optional<Role> servRole = roleService.findByName("ROLE_SERVICE");
         Optional<Role> modrRole = roleService.findByName("ROLE_MODERATOR");
 
+
```
достаточно одной пустой строки для логического разделения кода

tediik commented on src/main/java/com/jm/online_store/config/DataInitializer.java:
```diff
@@ -168,13 +168,19 @@ private void roleInit() {
+        deletedCustomer.setProfilePicture(StringUtils.cleanPath("deleted.jpg"));
+        deletedCustomer.setFirstName("Deleted");
+        deletedCustomer.setLastName("");
+
```
достаточно одной пустой строки для логического разделения кода

tediik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -95,8 +109,29 @@ public ResponseEntity changePassword(Model model,
     }
 
+    /**
+     * Метод который безвозвратно удаляет пользователя при нажатии на кнопку "удалить профиль" и
+     * сохраняет коммеатрии и отзывы под сущность DeletedCustomer
```
поправь ошибки в слове коммеатрии 
итак понятно, что это метод, поэтому начать лушче так
Безвозвратно удаляет пользователя при нажатии на кнопку "удалить профиль" и

tediik commented on src/main/java/com/jm/online_store/controller/rest/customer/CustomerRestController.java:
```diff
@@ -95,8 +109,29 @@ public ResponseEntity changePassword(Model model,
+    @DeleteMapping("/deleteProfileUnrecoverable/{id}")
+    @ApiOperation(value = "Delete Users unrecoverable, when Delete button clicked")
+    public ResponseEntity<String> deleteProfileUnrecoverable(@PathVariable Long id) {
+        User customer = userService.findUserById(id);
+        List<Comment> customerComments = commentService.findAllByCustomer(customer);
+        List<Review> customerReview = reviewService.findAllByCustomer(customer);
+        User deletedUser = userService.findUserByEmail("deleted@mail.ru");
+        for (Comment comment : customerComments) comment.setCustomer(deletedUser);
+        for (Review review : customerReview) review.setCustomer(deletedUser);
+        customerService.deleteByID(id);
+        return ResponseEntity.ok("Delete profile");
```
из-за того, что эта бизнес-логика напрямую в контроллере и на основе анализа остального кода я делаю такой вывод - ты сделал свою собственную логику для удаления пользователя - только при нажатии твоей кнопки, а как быть с тем функционалом, когда пользователь удаляется по истечении 30 дней после того как нажал обычную кнопку Удалить профиль?? Этот существующий функционал тоже должен работать аналогичным образом!

MaksimovMA committed:
BugFix

MaksimovMA committed:
Merge branch 'dev' into refactor/delete_customer

tediik committed:
Update CustomerServiceImpl.java

MaksimovMA committed:
Добавиляю метод в CustomerService который меняет по идентификатору пользоваетля в коммеанариях и отзывах на "Deleted пользователя(DeletedCustomer).
Использую данный метод для удаления.
Переношу кнопку безвозвратного удаления пользователя в модальное окно удаления пользователя.
